test/plugin/base_test.rb

```diff
@@ -338,6 +338,81 @@ def test_structured_payload_log
     end
   end
 
+  def test_stackdriver_trace_format_compliant_trace
+    full_stackdriver_trace = \
+      'projects/some-proj/traces/1234567890abcdef1234567890abcdef'
+
+    [
+      {
+        config: APPLICATION_DEFAULT_CONFIG
+      },
+      {
+        config: 'enable_stackdriver_trace_autoformat true'
+      },
+      {
+        config: 'enable_stackdriver_trace_autoformat false'
+      }
+    ].each do |test_params|
+      new_stub_context do
+        setup_gce_metadata_stubs
+        setup_logging_stubs do
+          d = create_driver(test_params[:config])
+          d.emit(DEFAULT_TRACE_KEY => full_stackdriver_trace)
+          d.run
+
+          verify_log_entries(1, COMPUTE_PARAMS, 'jsonPayload') do |entry|
+            assert_equal full_stackdriver_trace, entry['trace']
+          end
+        end
+      end
+    end
+  end
+
+  def test_autoformat_enabled_with_stackdriver_trace_format_noncompliant_trace
+    non_stackdriver_trace_compliant_trace = '1234567890abcdef1234567890abcdef'
+
+    [
+      {
+        config: APPLICATION_DEFAULT_CONFIG
+      },
+      {
+        config: 'enable_stackdriver_trace_autoformat true'
+      }
+    ].each do |test_params|
+      new_stub_context do
+        setup_gce_metadata_stubs
+        setup_logging_stubs do
+          d = create_driver(test_params[:config])
+          d.emit(DEFAULT_TRACE_KEY => non_stackdriver_trace_compliant_trace)
+          d.run
+
+          verify_log_entries(1, COMPUTE_PARAMS, 'jsonPayload') do |entry|
+            expected_trace = \
+              "projects/#{d.instance.project_id}" \
```

Comment:
This is a change detector test (it replicates the code under test). Please use the literal value. Better yet, define two constants, one for the compliant trace, and one for the non-compliant trace.

Reply:
Yeah, instead of `d.instance.project_id`, we could just use `PROJECT`, which is defined in the `constants.rb` file.

Reply:
I was proposing the `FULL_STACKDRIVER_TRACE` and `NON_STACKDRIVER_TRACE`, same as you... If we define one in terms of the other, we can even use `FULL_STACKDRIVER_TRACE` as the expected value here...